contracts/message-based/PriceAware.sol

```diff
@@ -3,49 +3,40 @@
 pragma solidity ^0.8.2;
 
 import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
-import "@openzeppelin/contracts/access/Ownable.sol";
 
-contract PriceAware is Ownable {
+abstract contract PriceAware {
   using ECDSA for bytes32;
 
-  uint256 public maxDelay = 3 * 60;
-  address private trustedSigner;
+  uint256 public maxDelay = 3 * 60; // seconds
 
-  function getTrustedSigner() public view virtual returns (address) {
-    return trustedSigner;
-  }
-
-  function setMaxDelay(uint256 _maxDelay) external onlyOwner {
-    maxDelay = _maxDelay;
-  }
-
-  function authorizeSigner(address _trustedSigner) external onlyOwner {
-    require(_trustedSigner != address(0));
-    trustedSigner = _trustedSigner;
+  /* ========== VIRTUAL FUNCTIONS (CAN BE OVERRIDEN) ========== */
 
-    emit TrustedSignerChanged(trustedSigner);
-  }
+  // This function must be implemented in child contract
+  function isSignerAuthorized(address _receviedSigner) internal virtual view returns (bool);
```

Comment:
Let's create a method isTimestampValid(uint256 timestamp),
We'd be able to put additional custom logic there or utilise CONSTANTS for gas savings.

Reply:
I've added function `validateTimestamp` instead.
It allows to pass different error messages

`  function validateTimestamp(uint256 _receivedTimestamp) internal virtual view {
    require(
      block.timestamp > _receivedTimestamp,
      "Data with future timestamps is not allowed");
    require(
      block.timestamp - _receivedTimestamp < MAX_DELAY,
      "Data is too old");
  }`